distributed/tests/test_client.py

```diff
@@ -165,6 +165,46 @@ def test_Future_exception_sync(loop):
             assert x.exception() is None
 
 
+@gen_cluster(client=True)
+def test_Future_release(c, s, a, b):
+    # Released Futures should be removed timely from the Client
+    x = c.submit(div, 1, 1)
+    yield x
+    x.release()
+    yield gen.moment
+    assert not c.futures
+
+    x = c.submit(slowinc, 1, delay=0.5)
+    x.release()
+    yield gen.moment
+    assert not c.futures
+
+    x = c.submit(div, 1, 0)
+    yield x.exception()
+    x.release()
+    yield gen.moment
+    assert not c.futures
+
+
+def test_Future_release_sync(loop):
+    # Released Futures should be removed timely from the Client
+    with cluster() as (s, [a, b]):
+        with Client(s['address'], loop=loop) as c:
+            x = c.submit(div, 1, 1)
+            x.result()
+            x.release()
+            wait_for(lambda: not c.futures, timeout=0.1)
+
+            x = c.submit(slowinc, 1, delay=0.5)
+            x.release()
+            wait_for(lambda: not c.futures, timeout=0.1)
+
+            x = c.submit(div, 1, 0)
+            x.exception()
+            x.release()
+            wait_for(lambda: not c.futures, timeout=0.1)
```

Comment:
I recommend increasing these timeouts for travis-ci.  It can occasionally be surprisingly slow.

Reply:
Ok, will do.